### PR TITLE
RxSwift 5.0.0 support

### PR DESCRIPTION
**Pull request by rynecheow**
## Prerequisite

Requires #145 to be merged and released first.

## Change
* Support macOS and watchOS through Carthage
* Added Example projects for macOS and watchOS
* Uses RxSwift 5.0.0
* Upgrade project to Swift 5.0

### CODE REVIEW

**Comment by rynecheow** [comment]
@freak4pc unit tests are failing due to https://github.com/CocoaPods/CocoaPods/issues/8729

**Comment by freak4pc** [comment]
That's very strange. I'm wondering how the tests on the main repo work with no issues (I mean on ReactiveX/RxSwift).

**Comment by rynecheow** [comment]
@freak4pc Do you have any idea?

**Comment by freak4pc** [comment]
No, I really don't at this moment. I'll try to take a look tomorrow when I have some free time.

**Comment by rynecheow** [comment]
Any luck?

File: .circleci/config.yml [hunk 1]
```diff
@@ -24,12 +24,9 @@ jobs:
           key: v1-dep-{{ .Branch }}-{{ epoch }}
           paths:
             - Carthage
-      - run:
-          name: Run Tests (Swift 4.2)
```
Comment (freak4pc) [inline .circleci/config.yml]: We're still supporting 4.2, just under the Swift 5 compiler.

**Comment by freak4pc** [comment]
I was sure I got it to work but it's still failing. Super weird because it works fine for other projects, I'm not sure what's broken in the project structure here but it might be better to try and figure it out or rebuild the example project 🤔

**Comment by rynecheow** [comment]
@freak4pc I've moved the unit tests away from the Cocoapods examples into the main project file - so that now it resolves correctly. Not sure if this deviates from your original intention of having the tests in the cocoapods integrated workspace..

File: .circleci/config.yml [hunk 2]
```diff
@@ -13,23 +13,20 @@ jobs:
...
-            - v1-dep-{{ .Branch }}-
-            - v1-dep-master-
-            - v1-dep-
-      - run: 
-          name: Bootstrap Carthage
-          command: scripts/bootstrap-if-needed.sh
+          key: RxAlamofire-Test-Carthage-Dependency
+      - run:
+          name: Install Carthage
+          command: set -o pipefail && carthage update --cache-builds --new-resolver --configuration Debug
```
Comment (freak4pc) [inline .circleci/config.yml]: Very interesting! I didn't know `--cache-builds` and `--new-resolver` are a thing. Mind sharing a link what they do ? Out of curiosity mainly.

**Pull request merged**